### PR TITLE
[Debugger] Clarify that only one paused task can be debugged at a time

### PR DESCRIPTION
**Pull request by pcmoritz**
## Why are these changes needed?

I was initially confused that I couldn't join another paused task while a debugger was in "continue" mode.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/ray-observability/ray-distributed-debugger.rst [hunk 1]
```diff
@@ -77,10 +77,14 @@ When the debugger hits a breakpoint:
     :align: center
 
 
-Start debugging
-~~~~~~~~~~~~~~~
+Start and stop debugging
+~~~~~~~~~~~~~~~~~~~~~~~~
 
-Debug your Ray app as you would when developing locally.
+Debug your Ray app as you would when developing locally. After you are done debugging this particular
```
Comment (angelinalg) [inline doc/source/ray-observability/ray-distributed-debugger.rst:83]:
```suggestion
Debug your Ray app as you would when developing locally. After you're done debugging this particular
```

File: doc/source/ray-observability/ray-distributed-debugger.rst [hunk 2]
```diff
@@ -77,10 +77,14 @@ When the debugger hits a breakpoint:
...
+breakpoint, click the "Disconnect" button in the debugging toolbar so you can join another task
```
Comment (angelinalg) [inline doc/source/ray-observability/ray-distributed-debugger.rst:84]:
```suggestion
breakpoint, click the **Disconnect** button in the debugging toolbar so you can join another task
```

**Review by angelinalg — COMMENTED** [review]
Mind making these style changes?

**Comment by pcmoritz** [comment]
Updated :)

File: doc/source/ray-observability/ray-distributed-debugger.rst [hunk 3]
```diff
@@ -77,10 +77,14 @@ When the debugger hits a breakpoint:
...
 
-Start debugging
-~~~~~~~~~~~~~~~
+Start and stop debugging
+~~~~~~~~~~~~~~~~~~~~~~~~
 
-Debug your Ray app as you would when developing locally.
+Debug your Ray app as you would when developing locally. After you're done debugging this particular
+breakpoint, click the **Disconnect** button in the debugging toolbar so you can join another task
+in the "Paused Tasks" list.
```
Comment (angelinalg) [inline doc/source/ray-observability/ray-distributed-debugger.rst:85]:
```suggestion
in the **Paused Tasks** list.
```

Comment (angelinalg) [reply doc/source/ray-observability/ray-distributed-debugger.rst:85]: I think you missed this one.

**Review by angelinalg — APPROVED** [review]

**Comment by scottsun94** [comment]
Screenshot 2024-10-30 at 11 05 16 AM
Is there a way to add a tooltip saying something like `Unable to start because you are debugging another task`


> Would be nice if we can disconnect automatically after clicking continue and the task is done, but this will require more investigation

Any plan for this?

**Comment by pcmoritz** [comment]
The tooltip would be great! @brycehuang30 feel free to put it on the backlog :)

The automatic disconnect most likely has lower ROI and would only be worth it if we are very sure that's what the user wants (and it is not brittle).

**Pull request merged**